contracts/src/LightClient.sol

```diff
@@ -341,4 +381,75 @@ contract LightClient is Initializable, OwnableUpgradeable, UUPSUpgradeable {
             revert NoChangeRequired();
         }
     }
+
+    /// @notice checks the stateUpdateBlockNumbers array and the delayThreshold to determine if
+    /// hotshot was
+    /// down at a specified L1 block number
+    /// @param l1BlockNumber This L1 block number used to reference a point in time when this light
+    /// client contract was expected to be updated in a given threshold
+    /// @param delayThreshold The delay threshold used to determined if this Light Client contract
+    /// was updated in the desired time (measured in blocks)
+    function wasL1Updated(uint256 l1BlockNumber, uint256 delayThreshold)
```

Comment:
I propose to invert the meaning of this function because I think it makes it clearer what it is for and easier to explain the logic. Currently this function is queried to turn on the escape hatch if the function returns false. With the change the function is called to turn on the escape hatch if the function returns true.

```solidity
    /// @notice checks that no more than threshold blocks passed since the last update before
    /// l1BlockNumber
    /// @param blockNumber The L1 block number
    /// @param threshold The number of blocks updates to this contract is allowed to lag behind
    function lagOverThreshold(uint256 blockNumber, uint256 threshold) public view returns (bool) {
        uint256 latestUpdateBefore;
        bool updateFound;
        // Iterate over updates in revers order to find the latest update before the given block
        // number.
        uint256 index = stateUpdateBlockNumbers.length - 1;
        while (!updateFound) {
            if (stateUpdateBlockNumbers[index] <= blockNumber) {
                updateFound = true;
                latestUpdateBefore = stateUpdateBlockNumbers[index];
            }
            if (index == 0) {
                break;
            }
            index--;
        }

        // If no snapshot is found, we don't have enough history stored to tell whether HotShot was
        // down.
        if (!updateFound) {
            revert InsufficientSnapshotHistory();
        }

        return blockNumber - latestUpdateBefore > threshold;
    }
```

thoughts?

cc @alysiahuggins @ImJeremyHe @nomaxg

Reply:
i like this too actually - It matches with the mindset of the person who's calling the function and wants to determine if they need to do some action in event of a lag

Reply:
make sense to me